### PR TITLE
chore: Add documentation for dynamo.compile backend

### LINKED ISSUE #2378: Documentation on dynamo.compile frontend usage

### PR DESCRIPTION
**Pull request by peri044**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #2378 

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: docsrc/user_guide/dynamo_export.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
+.. _getting_started_with_dynamo:
```
Comment (gs-olive) [inline docsrc/user_guide/dynamo_export.rst:1]: To be reflected in the index's `ref`, this would need to be changed to `_dynamo_export`

File: docsrc/user_guide/dynamo_export.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+There are lot of options for users to customize their settings for optimizing with TensorRT. 
+Some of the frequently used options are as follows: 
+
+
+* inputs - For static shapes, this can be a list of torch tensors or `torch_tensorrt.Input` objects. For dynamic shapes, this should be a list of ``torch_tensorrt.Input`` objects.
+* enabled_precisions - Set of precisions that TensorRT builder can use during optimization.
+* truncate_long_and_double - Truncates long and double values to int and floats respectively.
+* torch_executed_ops - Operators which are forced to be executed by Torch.
+* min_block_size - Minimum number of consecutive operators required to be executed as a TensorRT segment.
```
Comment (gs-olive) [inline docsrc/user_guide/dynamo_export.rst:36]: Consider adding backticks (``) to each option name argument to format as code

File: docsrc/user_guide/dynamo_export.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+usage of the dynamo backend  
+
+.. code-block:: python
+
+    import torch
+    import torch_tensorrt
+
+    model = MyModel().eval().cuda()
+    inputs = torch.randn((1, 3, 224, 224), dtype=torch.float32).cuda()
+    exp_program = torch.export(model, inputs)
```
Comment (gs-olive) [inline docsrc/user_guide/dynamo_export.rst:18]: Switch to `torch.export.export` for `2.1` ([reference](https://pytorch.org/docs/stable/export.html))

Comment (gs-olive) [reply docsrc/user_guide/dynamo_export.rst:18]: Additionallty, `inputs` should be a tuple of arguments to agree with the [API recommendation](https://github.com/pytorch/pytorch/blob/192477b5ba991b5e160d7264230945b49695c0e0/torch/export/__init__.py#L332)

File: docsrc/user_guide/dynamo_export.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    trt_gm = torch_tensorrt.dynamo.compile(exp_program, inputs) # Output is a torch.fx.GraphModule
+    trt_gm(inputs)
```
Comment (gs-olive) [inline docsrc/user_guide/dynamo_export.rst:20]: These could need modification if the above `inputs` is changed to a tuple

**Review by gs-olive — CHANGES_REQUESTED** [review]
Added a few comments on documentation improvements for this PR.

As part of this PR, could you also fix the `.rst` linking in `index.rst` for the `saving_models` file (see here in _User Guide_ for current rendering: https://pytorch.org/TensorRT/)

File: docsrc/user_guide/dynamo_export.rst (hunk 4 above)
Comment (peri044) [reply docsrc/user_guide/dynamo_export.rst]: done
